core/store/src/trie/mod.rs

```diff
@@ -384,7 +385,7 @@ impl RawTrieNode {
 impl RawTrieNodeWithSize {
     fn encode_into(&self, out: &mut Vec<u8>) -> Result<(), std::io::Error> {
         self.node.encode_into(out)?;
```

Comment:
Actually, similarly here -- I don't see how encoding can be a failable operation. The only thing that can fail here is memory allocation, but Rust aborts on OOM.